### ISSUE TITLE
Upgrade 20.0.0.5 → 20.0.0.7 blocked because "Updates for apps are not available"

### ISSUE THREAD

**Issue by frederikmoellers** [nextcloud/server#25603]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install Nextcloud 20.0.0.5
2.  Install (at least one of) the following apps: Deck, Music, News, Two-Factor TOTP Provider, Two-Factor U2F
3. Check the settings for available upgrades

### Expected behaviour
Nextcloud should recognize that these apps are available for 20.0.0.7 and offer to upgrade without issues.

### Actual behaviour
Nextcloud says that an update to 20.0.0.7 is available, but that updates are missing for the aforementioned apps: Deck, Music, News, Two-Factor TOTP Provider, Two-Factor U2F

### Server configuration
<details>
<summary>Server config</summary>
Operating system: Linux 4.19.0-13-amd64 #1 SMP Debian 4.19.160-2 (2020-11-28) x86_64

Webserver: Apache/2.4.38 (Debian) (apache2handler)

Database: mysql 10.3.27

PHP version: 7.3.19-1~deb10u1

Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, sodium, standard, apache2handler, mysqlnd, PDO, xml, apcu, bcmath, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, gmp, iconv, imagick, intl, json, ldap, exif, mysqli, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, apc, posix, pspell, readline, shmop, SimpleXML, soap, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlrpc, xmlwriter, xsl, zip, Phar, Zend OPcache

Nextcloud version: 20.0.5 - 20.0.5.2

Updated from an older Nextcloud/ownCloud or fresh install:

Where did you install Nextcloud from: Server Tarball from nextcloud.com

<details><summary>Signing status</summary>

Array ( )

</details>

<details><summary>List of activated apps</summary>

Enabled:
 - accessibility: 1.6.0
 - activity: 2.13.4
 - calendar: 2.1.3
 - cloud_federation_api: 1.3.0
 - comments: 1.10.0
 - contacts: 3.4.3
 - contactsinteraction: 1.1.0
 - cospend: 1.2.9
 - dashboard: 7.0.0
 - dav: 1.16.2
 - deck: 1.2.3
 - federatedfilesharing: 1.10.2
 - federation: 1.10.1
 - files: 1.15.0
 - files_automatedtagging: 1.10.1
 - files_external: 1.11.1
 - files_lock: 20.0.0
 - files_markdown: 2.3.1
 - files_pdfviewer: 2.0.1
 - files_rightclick: 0.17.0
 - files_sharing: 1.12.2
 - files_trashbin: 1.10.1
 - files_versions: 1.13.0
 - files_videoplayer: 1.9.0
 - firstrunwizard: 2.9.0
 - flow_notifications: 1.0.3
 - forms: 2.1.0
 - health: 0.3.0
 - impersonate: 1.7.0
 - logreader: 2.5.0
 - lookup_server_connector: 1.8.0
 - maps: 0.1.8
 - music: 0.17.3
 - news: 15.1.1
 - nextcloud_announcements: 1.9.0
 - notes: 4.0.2
 - notifications: 2.8.0
 - oauth2: 1.8.0
 - password_policy: 1.10.1
 - phonetrack: 0.6.5
 - photos: 1.2.3
 - privacy: 1.4.0
 - provisioning_api: 1.10.0
 - recommendations: 0.8.0
 - serverinfo: 1.10.0
 - settings: 1.2.0
 - sharebymail: 1.10.0
 - social: 0.4.2
 - socialsharing_email: 2.1.0
 - spreed: 10.0.5
 - support: 1.3.0
 - survey_client: 1.8.0
 - systemtags: 1.10.0
 - tasks: 0.13.6
 - text: 3.1.0
 - theming: 1.11.0
 - timetracker: 0.0.53
 - twofactor_backupcodes: 1.9.0
 - twofactor_totp: 5.0.0
 - twofactor_u2f: 6.0.0
 - updatenotification: 1.10.0
 - user_status: 1.0.1
 - viewer: 1.4.0
 - weather_status: 1.0.0
 - workflow_script: 1.5.1
 - workflowengine: 2.2.0
Disabled:
 - admin_audit
 - encryption
 - officeonline
 - onlyoffice
 - user_ldap
 - workflow_ocr

</details>

<details><summary>Configuration (config/config.php)</summary>

{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "nextcloud.die-sinlosen.de",
        "172.18.0.1"
    ],
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "https:\/\/nextcloud.die-sinlosen.de",
    "dbtype": "mysql",
    "version": "20.0.5.2",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "UTC",
    "installed": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "mail_smtpmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false,
    "loglevel": 0,
    "trashbin_retention_obligation": "auto, 365",
    "version_retention_obligation": "auto, 365",
    "updatechecker": true,
    "htaccess.RewriteBase": "\/",
    "singleuser": false,
    "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "25",
    "mysql.utf8mb4": true,
    "theme": "",
    "upgrade.disable-web": "false",
    "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
    "mail_sendmailmode": "smtp",
    "updater.release.channel": "stable"
}

</details>

External storages: yes

<details><summary>External storage configuration</summary>

No mounts configured

</details>

Encryption: no

User-backends:

OC\User\Database

Browser: Mozilla/5.0 (X11; Linux x86_64; rv:85.0) Gecko/20100101 Firefox/85.0

### Logs

<details>
<summary>Server log</summary>

```
No entries that are related
```
</details>

#### Web server error log
<details>
<summary>Web server error log</summary>

```
No entries that are related
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
No entries that are related
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
No entries that are related
```
</details>
</details>

**Comment by eibex** [comment]
20.0.6 to 20.0.7 showed the same warning but it is a false positive as apps did not get removed or disabled during the upgrade: https://github.com/nextcloud/twofactor_totp/issues/1073

I am not completely sure it is the same for 20.0.5, but chances are it is.

**Comment by szaimen** [comment]
Yes, this is due to a bug in 20.0.5 and 20.0.6. this was fixed with 20.0.7

**Comment by frederikmoellers** [comment]
OK thanks for the info! 👍
